src/MSBuild/XMake.cs

```diff
@@ -1056,6 +1060,7 @@ string outputResultsCache
                 ToolsetDefinitionLocations toolsetDefinitionLocations = ToolsetDefinitionLocations.Default;
 
                 bool preprocessOnly = preprocessWriter != null && !FileUtilities.IsSolutionFilename(projectFile);
+                bool targetsOnly = targetsWriter != null && !FileUtilities.IsSolutionFilename(projectFile);
```

Comment:
This does not seem logical for me at first glance, because printing the targets should be a read-only operation.

Also, it looks like the setting will be ignored anyway (see [this comment](https://github.com/Microsoft/msbuild/blob/08d1cb5532d98dd4b9f2f4ecdb7c9c7a037aed1c/src/Build/Xml/XmlReaderExtension.cs#L83-L85)).

Reply:
I thought I was suggesting it should be read-only, but I might have gotten it flipped. @rainersigwald, do you know better? Otherwise, this looks good to me.